pluma/cli/deviceactionregistry.py

```diff
@@ -1,20 +1,22 @@
+from typing import Dict, Optional, Type
+
 from pluma import Board
 from pluma.cli import DeviceActionBase
 
 
 class DeviceActionRegistry():
-    action_classes_dict = {}
+    action_classes_dict: Dict[str, Type[DeviceActionBase]] = {}
 
     @classmethod
     def register(cls, action_key: str):
-        def decorator(action_class: DeviceActionBase):
+        def decorator(action_class: Type):
             cls.register_class(action_class=action_class,
                                action_key=action_key)
             return action_class
         return decorator
 
     @classmethod
-    def register_class(cls, action_class: DeviceActionBase, action_key: str):
+    def register_class(cls, action_class: Type, action_key: str):
```

Comment:
You can specialize `Type` to indicate that `action_class` should be or inherit from `DeviceActionBase`:
```suggestion
    def register_class(cls, action_class: Type[DeviceActionBase], action_key: str):
```

Note that if you want to keep the subclass error check you'll need to replace `action_class.__name__` with `cast(Type, action_class).__name__` to make the type system happy.

Reply:
Done